Packages/Status/Sources/Status/Row/StatusRowViewModel.swift

```diff
@@ -288,28 +289,54 @@ public class StatusRowViewModel: ObservableObject {
   }
 
   func translate(userLang: String) async {
-    do {
-      withAnimation {
-        isLoadingTranslation = true
-      }
-      // We first use instance translation API if available.
-      let translation: StatusTranslation = try await client.post(endpoint: Statuses.translate(id: finalStatus.id,
-                                                                                              lang: userLang))
-      withAnimation {
-        self.translation = translation
-        isLoadingTranslation = false
-      }
-    } catch {
-      // If not or fail we use Ice Cubes own DeepL client.
-      let deepLClient = DeepLClient()
-      let translation = try? await deepLClient.request(target: userLang,
-                                                       source: finalStatus.language,
-                                                       text: finalStatus.content.asRawText)
-      withAnimation {
-        self.translation = translation
-        isLoadingTranslation = false
-      }
+    let userAPIKey = DeepLUserAPIHandler.readAndUpdate()
+    if userAPIKey == nil {
+      do {
+        withAnimation {
+          isLoadingTranslation = true
+        }
+        // We first use instance translation API if available.
+        let translation: StatusTranslation = try await client.post(endpoint: Statuses.translate(id: finalStatus.id,
+                                                                                                lang: userLang))
+        withAnimation {
+          self.translation = translation
+          isLoadingTranslation = false
+        }
+        
+        return
+      } catch {}
     }
+      
+    // If not or fail we use Ice Cubes own DeepL client.
+    await translateWithDeepL(userLang: userLang, userAPIKey: userAPIKey)
```

Comment:
Could this also be called in the catch like before? If the user don't set their own API Key it'll use mine right? So this should be fine like before.

Reply:
It'll use yours. 
I don't think it's needed to call translateWithDeepL in the catch block since we already call that function after we finished the if statement (if the return statement doesn't get executed). If the request to the instance fails, the empty catch is executed, then the code after the catch, which includes a call to translateWithDeepL (that call is only not executed if the return statement in the if statement (ln 306) is reached, which happens if the instance request is successful).